### PR TITLE
esp32/modnetwork: Add esp_mac method, to get and set mac address

### PR DESCRIPTION
**Pull request by luaks**

### CODE REVIEW

File: esp32/modnetwork.c [hunk 1]
```diff
@@ -96,12 +96,13 @@ static inline void esp_exceptions(esp_err_t e) {
 
 typedef struct _wlan_if_obj_t {
     mp_obj_base_t base;
-    int if_id;
+    wifi_interface_t if_id;
+    uint8_t *mac;
 } wlan_if_obj_t;
 
 const mp_obj_type_t wlan_if_type;
-STATIC const wlan_if_obj_t wlan_sta_obj = {{&wlan_if_type}, WIFI_IF_STA};
```
Comment (luaks) [inline esp32/modnetwork.c:103]: This is one possible approach to store general configuration, like ssids, the mac address, auth configuration. Pycom does it this way in their modwlan imlementation.

Another way would be to always get and set this information via the [esp_wifi_get_mac](https://esp-idf.readthedocs.io/en/latest/api/esp_wifi.html#_CPPv216esp_wifi_get_mac16wifi_interface_tA6_7uint8_t) and [esp_wifi_set_mac](https://esp-idf.readthedocs.io/en/latest/api/esp_wifi.html#_CPPv216esp_wifi_set_mac16wifi_interface_tA6_7uint8_t).

**Comment by dpgeorge** [comment]
Thanks for the contribution!

The aim for us here is to make the user-facing API for the ESP32 port as similar as possible to the ESP8266, so that the same code can run on both.  As such, the way to get/set the mac address would be via the wlan.config() method, ie wlan.config(mac=b'12341234') and wlan.config('mac').

Apart from that, let's wait for @nickzoic to merge the modsocket branch into the main esp32 branch before adding more features.

**Comment by luaks** [comment]
Oh, I have oriented my development on the documentation for the [Hardware API](https://github.com/micropython/micropython/wiki/Hardware-API#the-wlan-class).

It seems that there are multiple APIs documented. Is there any place where the APIs wanted for the esp32 are defined?
I'd love to help with porting micropython to the esp32, but without the knowledge of what is needed, it is rather hard.

**Comment by dpgeorge** [comment]
> It seems that there are multiple APIs documented. Is there any place where the APIs wanted for the esp32 are defined?

Yes, please look here: http://docs.micropython.org/en/latest/esp8266/library/network.html .  We want the esp32 to be as similar as possible to the esp8266, so go by those docs as much as is possible.

**Comment by nickzoic** [comment]
G'day Luaks, yeah I've implemented parts of network module but haven't filled in config() yet because I got distracted over in modsocket :-) ... if you can help out that's brilliant.  You could probably just start by grabbing extmod/modlwip.c's version and adapting it one chunk at a time.

**Comment by dpgeorge** [comment]
wlan.config() was added in 41534f18ef973ac6dc87c748fc1de1a9b2b25c7f.

**Pull request closed without merging**